delta-app/src/main/java/io/cdap/delta/store/StateStoreMigrator.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.delta.store;
+
+import io.cdap.delta.api.DeltaPipelineId;
+import io.cdap.delta.app.DeltaWorkerId;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+
+import java.io.IOException;
+
+/**
+ * A util class to transfer any state store data present on HCFS path by an old pipeline with HCF based state store
+ * into SQL based state store on the first pipeline run after upgrade.
+ */
+public class StateStoreMigrator {
+
+  private final String hcfsStateStorePath;
+  private final DeltaWorkerId currentId;
+  private final StateStore remoteStateStore;
+
+  private FileSystem fileSystem;
+  private Path instanceBasePath;
+  protected HCFSStateStore hcfsStateStore;
+  protected DeltaWorkerId previousId;
+
+  public StateStoreMigrator(String hcfsStateStorePath, DeltaWorkerId currentId, StateStore remoteStateStore) {
+    this.hcfsStateStorePath = hcfsStateStorePath;
+    this.currentId = currentId;
+    this.remoteStateStore = remoteStateStore;
+  }
+
+  public void perform() throws IOException {
```

Comment:
Maybe better to return boolean from this method to indicate whether we did any migration as a part of this call. in worker where we call this method then we can log the output as DEBUG.